lib/mutant/isolation.rb

```diff
@@ -33,30 +37,41 @@ module Fork
       #   if block terminates abnormal
       #
       # rubocop:disable MethodLength
-      def self.call(&block)
-        IO.pipe(binmode: true) do |reader, writer|
+      def call(&block)
+        io.pipe(binmode: true) do |reader, writer|
           writer.binmode
           begin
-            pid = Process.fork do
-              File.open(File::NULL, File::WRONLY) do |file|
-                $stderr.reopen(file)
-                reader.close
-                writer.write(Marshal.dump(block.call))
-                writer.close
-              end
+            pid = process.fork do
+              child(reader, writer, &block)
             end
 
             writer.close
-            Marshal.load(reader.read)
+            marshal.load(reader.read)
```

Comment:
I think you can pass in an `IO` object into `Mashall#load`, so you could write this as: `marshal.load(reader)`.